artichoke-backend/src/extn/core/kernel/integer.rs

```diff
@@ -0,0 +1,177 @@
+    let arg = args.arg.as_str();
+    let radix = args.radix;
+    let raise_exception = args.raise_exception;
+
+    let mut string = String::new();
+
+    // If have mutliple consecutive embedded underscores, argument error!
+    let multi_underscore_re = Regex::new(r"__+").unwrap();
+    if arg.starts_with('_') || arg.ends_with('_') || multi_underscore_re.is_match(arg) {
+        return Err(Error::InvalidValue(arg.into(), raise_exception));
+    }
+
+    // If have multiple consecutive leading/trailing signs, argument error!
+    let multi_sign_re = Regex::new(r"\+\++|\-\-+").unwrap();
+    if multi_sign_re.is_match(arg) {
+        return Err(Error::InvalidValue(arg.into(), raise_exception));
+    }
+
+    // Remove embedded underscore `_`
+    // because they represent error in `from_str_radix` & `from_str`
+    let arg = arg.replace('_', "");
+
+    // Remove leading and trailing white space,
+    // because they represent error in `from_str_radix` & `from_str`.
+    let arg = arg.trim();
+    let arg = if arg.starts_with('-') {
+        string.push('-');
+        &arg[1..]
+    } else if arg.starts_with('+') {
+        string.push('+');
+        &arg[1..]
+    } else {
+        arg
+    };
+
+    // if `arg` is null byte, raise `ArgumentError`
+    if arg.contains('\0') {
+        return Err(Error::ContainsNullByte(raise_exception));
+    }
+
+    let mut parsed_radix = None;
+    if arg.starts_with('0') && arg.len() > 2 {
+        match &arg[0..2] {
+            "0b" | "0B" => {
+                string.push_str(&arg[2..]);
+                parsed_radix = Some(2);
+            }
+            "0o" | "0O" => {
+                string.push_str(&arg[2..]);
+                parsed_radix = Some(8);
+            }
+            "0d" | "0D" => {
+                string.push_str(&arg[2..]);
+                parsed_radix = Some(10);
+            }
+            "0x" | "0X" => {
+                string.push_str(&arg[2..]);
+                parsed_radix = Some(16);
+            }
+            prefix if &prefix[0..1] == "0" => {
+                string.push_str(&arg[1..]);
+                parsed_radix = Some(8);
+            }
+            _ => {}
+        };
+    } else {
+        string.push_str(arg);
+    }
+
+    match (radix, parsed_radix) {
+        (Some(radix), Some(parsed_radix)) => {
+            if radix != parsed_radix {
+                return Err(Error::InvalidValue(string, raise_exception));
+            }
+            if let Ok(v) = i64::from_str_radix(string.as_str(), radix as u32) {
+                Ok(interp.convert(v))
+            } else {
+                Err(Error::InvalidValue(string, raise_exception))
+            }
+        }
+        (Some(radix), None) | (None, Some(radix)) => {
+            if radix < 2 || radix > 36 {
+                return Err(Error::InvalidRadix(radix.to_string(), raise_exception));
+            }
+            if let Ok(v) = i64::from_str_radix(string.as_str(), radix as u32) {
```

Comment:
not all platforms uses `i64` as the `Integer` type.

There is a type alias defined at `crate::types::Int`.